aepp/exportDatasetToDataLandingZone.py

```diff
@@ -0,0 +1,390 @@
+        )
+        flow_obj = asdict(flow_data)
+        flow_res = self.flow_conn.createFlow(
+            obj = flow_obj,
+            flow_spec_id = self.DLZ_FLOW_SEPC_ID
+        )
+        try:
+            dataflow_id = flow_res["id"]
+            if self.loggingEnabled:
+                self.logger.info("dataflowId " + dataflow_id + " has been created")
+            return dataflow_id
+        except KeyError:
+            raise RuntimeError(f"Error when creating data flow: {flow_res}")
+
+    def createFlowRun(self, dataflow_id: str, dataset_id: str, on_schedule: bool, initial_delay: int = 300):
+        """
+        Create a data flow run
+        Arguments:
+          dataflow_id : REQUIRED : The flow that needs to be triggered a flow run
+          dataset_id : REQUIRED : The dataset that needs to be exported
+          on_schedule: REQUIRED: define whether you would like to have the export following a fixed schedule or not
+        """
+        if not on_schedule:
+            #set up initial delay due to hollow refresh time interval
+            if self.loggingEnabled:
+                self.logger.info(f"Waiting for flow {dataflow_id} to refreshed after {initial_delay} seconds")
+            time.sleep(initial_delay)
+            if self.loggingEnabled:
+                self.logger.info(f"Start create adhoc dataset export for flow {dataflow_id} with dataset {dataset_id }")
+            res = self.retryOnNotReadyException(dataflow_id = dataflow_id, dataset_id= dataset_id)
+            flow_run_id = res["destinations"][0]["datasets"][0]["statusURL"].rsplit('/', 1)[-1]
+            #check if flowRun has finished
+            finished = False
+            while not finished:
+                try:
+                    run = self.flow_conn.getRun(runId = flow_run_id)["items"][0]
+                    run_id = run["id"]
+                    run_started_at = run["metrics"]["durationSummary"]["startedAtUTC"]
+                    run_ended_at = run["metrics"]["durationSummary"]["completedAtUTC"]
+                    run_duration_secs = (run_ended_at - run_started_at) / 1000
+                    run_size_mb = run["metrics"]["sizeSummary"]["outputBytes"] / 1024. / 1024.
+                    run_num_rows = run["metrics"]["recordSummary"]["outputRecordCount"]
+                    run_num_files = run["metrics"]["fileSummary"]["outputFileCount"]
+                    if self.loggingEnabled:
+                        self.logger.info(f"Run ID {run_id} completed with: duration={run_duration_secs} secs; size={run_size_mb} MB; num_rows={run_num_rows}; num_files={run_num_files}")
+                    finished = True
+                except Exception as e:
+                    if self.loggingEnabled:
+                        self.logger.warn(f"No runs completed yet for flow {dataflow_id}")
+                    time.sleep(30)
+            #push the startDate to a year from now
+            start_time = int(time.time() + 60*60*24*365)
+            op = [
```

Comment:
here as well, should op be a dataclass as well

Reply:
Adding data class for above